### ISSUE TITLE
IN operator in kundera hbase

### ISSUE THREAD

**Issue by jeevitesh** [Impetus/kundera#633]
Hi,
I'm using kundera 2.10, along with HBASE
Issue is I have collection of ID (here in the form of List).
So need to get all the details in db for the list of ID's so planned to use IN operator.
Here is the code:

 List<String> names = Arrays.asList("00969425", "00969442");
            String jpql = "Select t from Data t where t.ID IN :names";
            Query query = entityManager.createQuery(jpql,Data.class);
            query.setParameter("names", names);
            List<Data> rows = query.getResultList();

```
          for(Data row :rows)
          {
             String rowID = row.getParentRowID();

          }
```

Here is the exception it say IN operator is not supported
SEVERE: The RuntimeException could not be mapped to a response, re-throwing to the HTTP container
java.lang.UnsupportedOperationException:  Condition IN is not suported in  hbase!

How to approach the above issue?.
Please need your valuable thoughts on the same.
Thanks
jeevitesh

**Comment by chhavigangwal** [comment]
@jeevitesh 

Currently Kundera doesn't support "In" clause for Hbase , so you can use range query to handle your case by adding exceptions to ids not required in given range. 

You can use the following test case for reference :

https://github.com/impetus-opensource/Kundera/blob/273c13342ddd1aceed0cd23504649926ce8fdb84/src/kundera-hbase/src/test/java/com/impetus/client/hbase/crud/PersonHBaseTest.java

Hope that Helps!
Chhavu

**Comment by chhavigangwal** [comment]
A fix for enabling "In" clause will be added in current trunk soon!

Chhavi

**Comment by chhavigangwal** [comment]
A fix for the same has been added in current trunk.

Chhavi

**Comment by chhavigangwal** [comment]
Fixed with Kundera -2.14 Release
